python/rapidsmpf/rapidsmpf/tests/streaming/test_pyobject.py

```diff
@@ -0,0 +1,157 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import weakref
+from typing import TYPE_CHECKING
+
+from rapidsmpf.streaming.core.channel import Channel, Message
+from rapidsmpf.streaming.core.leaf_node import pull_from_channel, push_to_channel
+from rapidsmpf.streaming.core.node import define_py_node, run_streaming_pipeline
+from rapidsmpf.streaming.core.pyobject import PyObjectPayload
+
+if TYPE_CHECKING:
+    from concurrent.futures import ThreadPoolExecutor
+
+    from rapidsmpf.streaming.core.context import Context
+
+
+def test_from_object_basic() -> None:
+    """Test creating PyObjectPayload from basic Python objects."""
+    # Test with dict
+    data = {"key": "value", "number": 42}
+    payload = PyObjectPayload.from_object(sequence_number=0, obj=data)
+    assert payload.sequence_number == 0
+    assert payload.extract_object() == data
+
+    # Test with list
+    data_list = [1, 2, 3, 4, 5]
+    payload_list = PyObjectPayload.from_object(sequence_number=1, obj=data_list)
+    assert payload_list.sequence_number == 1
+    assert payload_list.extract_object() == data_list
+
+    # Test with tuple
+    data_tuple = ("a", "b", "c")
+    payload_tuple = PyObjectPayload.from_object(sequence_number=2, obj=data_tuple)
+    assert payload_tuple.sequence_number == 2
+    assert payload_tuple.extract_object() == data_tuple
+
+
+def test_message_roundtrip() -> None:
+    """Test that PyObjectPayload can be wrapped in a message and extracted."""
+    original_data = {"user": "alice", "score": 100, "active": True}
+    payload1 = PyObjectPayload.from_object(sequence_number=42, obj=original_data)
+
+    # Wrap in message
+    msg = Message(payload1)
```

Comment:
Something's not very clear to me here. 
If this was a real send-recv scenario, where would the `payload1.extract_object()` be called? Its not sufficient to call `del payload1` here, isnt it? Because it's refcount has been incremented. I feel like there is a memory-leak in the send side.

Reply:
The `OwningWrapper`, when it goes out of scope, calls the deleter which decrefs the pointer.